packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3083,10 +3089,20 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   }
 
   void _replaceText(ReplaceTextIntent intent) {
-    userUpdateTextEditingValue(
-      intent.currentTextEditingValue.replaced(intent.replacementRange, intent.replacementText),
-      intent.cause,
+    final TextEditingValue oldValue = _value;
+    final TextEditingValue newValue = intent.currentTextEditingValue.replaced(
+      intent.replacementRange,
+      intent.replacementText,
     );
+    userUpdateTextEditingValue(newValue, intent.cause);
+
+    // If there's no change in text and selection (e.g. when selecting and
+    // pasting identical text), the widget won't be rebuilt on value update.
+    // Handle this by calling _didChangeTextEditingValue() so caret and scroll
+    // updates can happen.
```

Comment:
What corner case is this handling exactly?  Maybe consider adding a test that covers this (unless I misunderstand and the existing test does cover it).

Reply:
Imagine you type "Lorem ipsum", select "ipsum", copy it and paste it into the same selection. Theoretically this is user input so you'd like to scroll, reset the caret animation, update selection/toolbar overlays etc. But 
`_didChangeTextEditingValue` won't be called because:
- it's a listener on `TextEditingController` which is a `ValueNotifier<TextEditingValue>`
- `TextEditingValue` equality operator will see the same value (text, selection, composing)
- `ValueNotifier`'s value setter won't call `notifyListeners()`

The interesting thing is that the "Selection will be scrolled into view with SelectionChangedCause" test in `editable_text_test.dart` already caught that although perhaps it wasn't intentional. What it does is to set a (0, 1) selection on each call to resetSelectionAndScrollOffset() and then do 2x cut and 2x paste. So you always end up pasting the same character that's already there.

Reply:
In the example you gave, wouldn't there be a selection change?

  1. "Lorem [ipsum]"
  2. Paste "ipsum"
  3. "Lorem ipsum|"

Reply:
Just tried in a TextField - no selection change on paste. In MS Word though, selection changes to collapsed on pasted text end.

Reply:
https://github.com/flutter/flutter/issues/98677

Reply:
Great, thanks for submitting the two issues!

Reply:
Does it make sense to move this to the `pasteText` method?

Reply:
My intention was to avoid bundling side effects that are sometimes unexpected (such as this one) into generic operations like replacing text. It's a bit cleaner that way IMO especially when you are reading the code in `pasteText`, and you'll know immediately this is going to bring the new selection into view, instead of having to trace to `_replaceText` and find the "bring into view" logic there. But it's just a nit.